### DISCUSSION TITLE
Adding mass element at a distance

### DISCUSSION THREAD

**Discussion by rahultarale** [petrobras/ross#990] [General]
Hi,
I have a mass disc component overhanging on the rotor whose C.G location is not on the rotor. I have C.Cg distance, mass, Ip, Id of the disc. I want to model that disc as lumped mass connected to the end of the shaft from the C.G location. Please let me know the procedure how I can model it. I have added a representative image below. As shown in  below example I need to know how to model the connecting element between shaft and lumped mass

Also I want to know how can I add structural damping ratio or material damping while performing unbalance analysis.

![Shaft](https://github.com/petrobras/ross/assets/58654595/acc710fd-cfa7-4315-8cba-f698865616a3)
![image](https://github.com/petrobras/ross/assets/58654595/4aa54819-fb52-4cba-b4f6-8fcc0bf9a32c)

**Comment by raphaeltimbo** [comment]
Hi @rahultarale!
The connecting element in this case would be a `ShaftElement`. 
You would need to create a new material for this specific element so that it does not contribute mass to the system. The element will also need to have a high stiffness.

One alternative approach would be to model the disk itself as a `ShaftElement` and adjust dimensions and material density to achieve the desired values:

```python
import ross as rs

material = rs.Material(name="material", rho=7850, E=203.2e9, G_s=80e9)
sh = rs.ShaftElement(
    material=material,
    L=0.5,
    idl=0.0,
    odl=1.,
    idr=0.0,
    odr=0.15,
)

rotor = rs.Rotor(shaft_elements=[sh])
s = rotor.shaft_elements[0]
print(s.axial_cg_pos)
````

![image](https://github.com/petrobras/ross/assets/18506378/cf3c68bc-1b3d-45ed-89a9-13804e7baac5)

**Comment by rahultarale** [comment]
Thanks for your response.
